PocketKit/Sources/PocketKit/Premium/PremiumUpgradeView.swift

```diff
@@ -175,12 +176,51 @@ Refunds are not available for unused portions of a subscription.
 }
 
 private struct PremiumTermsView: View {
+    @State var showPrivacyPolicy = false
+    @State var showTermsOfService = false
+
     var body: some View {
         HStack(spacing: 16) {
-            Button { } label: { Text("Privacy Policy").style(.terms) }
-            Button { } label: { Text("Terms of Service").style(.terms) }
+            Button(action: {
+                self.showPrivacyPolicy = true
+            }, label: { Text("Privacy Policy").style(.terms) })
+            .sheet(isPresented: $showPrivacyPolicy) {
+                if let privacyUrl = getUrlFor(typeOf: .PrivacyPolicy) {
+                    SFSafariView(url: privacyUrl)
+                }
+            }
+            .accessibilityIdentifier("privacy-policy")
+            Button(action: {
+                self.showTermsOfService = true
+            }, label: { Text("Terms of Service").style(.terms) })
+            .sheet(isPresented: $showTermsOfService) {
+                if let toSUrl = getUrlFor(typeOf: .TermsOfService) {
+                    SFSafariView(url: toSUrl)
+                }
+            }
+            .accessibilityIdentifier("terms-of-service")
         }.padding(.bottom)
     }
+
+    enum Link {
+        case PrivacyPolicy
+        case TermsOfService
+    }
+
+    private func getUrlFor(typeOf: Link) -> URL? {
+        switch typeOf {
+        case .PrivacyPolicy:
+            guard let privacyUrl = URL(string: "https://getpocket.com/privacy/") else {
+                return nil
+            }
+            return privacyUrl
```

Comment:
can also do `return URL(string: "https://getpocket.com/privacy/")` instead of using the guard because we're return an optional. In fact, I'm okay with doing away with the enum entirely and just using 

```
if let privacyUrl = URL(string: "https://getpocket.com/privacy/") {
                     SFSafariView(url: privacyUrl)
                 }
                 ```

Reply:
General guard question here. We are defining the URL, I think Swfit's interface here is bad. I'd rather us force unwrap in this case and fail hard and fast if we are defining the input URL. No need to guard something we know will always be true.

Reply:
But if for some reason that URL is not accessible, we'd crash the app instead of just not showing the privacy, right?

Reply:
If we want to go through the failing hard and fast route, then I think we should capture the error. I personally don't like having force unwraps and rather have the button do nothing vs crash. However if we are okay using forced unwraps going forward for defined URLs, we can do that and I'll add it to our wiki.

It's very unlikely to crash, but still possible right?

Reply:
@Gio2018 

> But if for some reason that URL is not accessible, we'd crash the app instead of just not showing the privacy, right?

I don't believe that is the case, the URL does not make any network calls. It would crash only if the value passed to URL was not a valid URL pattern. In this case, the URL is defined very strictly and passed in by us. 

 
@cyndichin 
> It's very unlikely to crash, but still possible right? 
The only crash I see happening here would be a developer error. Which we want to fail fast, and sentry will pick up.